src/main/java/freiman/conway/GridComponent.java

```diff
@@ -0,0 +1,63 @@
+package freiman.conway;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+
+public class GridComponent extends JComponent {
+    private Grid grid;
+    private final int gridSpacing = 15;
+
+
+    public GridComponent(Grid grid) {
+        this.grid = grid;
+        addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                grid.put(e.getX() / gridSpacing, e.getY() / gridSpacing);
+                repaint();
+
+            }
+
+            @Override
+            public void mousePressed(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseReleased(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseEntered(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseExited(MouseEvent e) {
+
+            }
+        });
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+
+        //draw the boxes
+        for (int y = 0; y < grid.getHeight(); y++) {
+            for (int x = 0; x < grid.getWidth(); x++) {
+                if (grid.isAlive(x, y)) {
+                    g.setColor(Color.GREEN);
+                    g.fillRect(x * gridSpacing, y * gridSpacing, gridSpacing, gridSpacing);
+                } else {
+                    g.setColor(Color.LIGHT_GRAY);
+                    g.drawRect(x * gridSpacing, y * gridSpacing, gridSpacing, gridSpacing);
+                }
```

Comment:
Set the whole background to light grey, therefore you just need to draw green squares.

Reply:
<img width="747" alt="Screenshot 2024-09-14 at 11 46 15 PM" src="https://github.com/user-attachments/assets/18e8c639-c19f-4e5b-aa67-f3a74eff2a12">
Like this?

Reply:
Yes